File: 1-easy/problems/3417-zigzag-grid-traversal-with-skip/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def zigzagTraversal(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: List[int]
        """
        result = []
        for i in range(len(grid)):
            if i%2 == 0:
                result.extend(grid[i][j] for j in range(0, len(grid[0]), 2))
            else:
                result.extend(grid[i][j] for j in reversed(range(1, len(grid[0]), 2)))
        return result

File: 1-easy/problems/3417-zigzag-grid-traversal-with-skip/solution.ts
function zigzagTraversal(grid: any): boolean | number | string | any {
  """
  :type grid: List[List[int]]
  :rtype: List[int]
  """
  result = []
  for (i in Array.from({length: (grid}, (_, i) => i).length)) {
      if (i%2 == 0) {
          result.extend(grid[i][j] for j in Array.from({length: 0, (grid[0]}, (_, i) => i).length, 2))
      } else {
          result.extend(grid[i][j] for j in reversed(Array.from({length: 1, (grid[0]}, (_, i) => i).length, 2)))
  return result;
}

export { zigzagTraversal };
